sharding/collation.go

```diff
@@ -22,17 +25,39 @@ type CollationHeader struct {
 	proposerSignature []byte          //the proposer's signature for calculating collation hash.
 }
 
+// Hash takes the keccak256 of the collation header's contents.
+func (h *CollationHeader) Hash() (hash common.Hash) {
+	hw := sha3.NewKeccak256()
+	rlp.Encode(hw, h)
+	hw.Sum(hash[:0])
+	return hash
+}
+
+// ShardID is the identifier for a shard.
+func (h *CollationHeader) ShardID() *big.Int { return h.shardID }
+
+// Period the collation corresponds to.
+func (h *CollationHeader) Period() *big.Int { return h.period }
+
+// ChunkRoot of the serialized collation body.
+func (h *CollationHeader) ChunkRoot() *common.Hash { return h.chunkRoot }
+
 // Header returns the collation's header.
 func (c *Collation) Header() *CollationHeader { return c.header }
 
-// Transactions returns an array of tx's in the collation.
-func (c *Collation) Transactions() []*types.Transaction { return c.transactions }
+// Body returns the collation's byte body.
+func (c *Collation) Body() []byte { return c.body }
```

Comment:
Shouldn’t this return a pointer to the byte array?

Reply:
Why would it? In case it's nil?